### ISSUE TITLE
result replication issue about ir2mol

### ISSUE THREAD

**Issue by X1nyuLu** [rxn4chemistry/MultimodalAnalytical#8]
This is a great work! I am trying to replicate your work about translate infrared spectra to smiles. Although I was able to run through the code, the metrics at the end of the model's training are very poor, and I cannot reach the values in the article. I am not sure where I went wrong. I hope you could give me some guidance.

here is my bash script:
```bash
#!/bin/bash

export TOKENIZERS_PARALLELISM=False
export TRANSFORMERS_VERBOSITY=error

python -m analytical_fm.cli.training \
            working_dir=runs/ \
            job_name=pretraining \
            data_path=data/pretraining/ \
            splitting=random \
            data=ir/patches \
            data.IR.preprocessor_arguments.patch_size=75 \
            data.IR.preprocessor_arguments.interpolation=True \
            data.Formula.column=molecular_formula \
            model=custom_model \
            molecules=True \
            trainer.epochs=60 \
            model.lr=1e-3 \
            model.positional_encoding_type=learned \
            model.gated_linear=True \
            model.optimiser=adamw

```

here is the output log:
```text
[2025-07-03 12:42:03,665][__main__][INFO] - {'Formula': {'type': 'text', 'column': 'molecular_formula', 'target': False, 'preprocessor_arguments': {'tokenizer': 'formula', 'tokenizer_regex': '([A-Z]{1}[a-z]?[0-9]*)'}}, 'IR': {'type': '1D_patches', 'column': 'ir_spectra', 'target': False, 'preprocessor_arguments': {'patch_size': 75, 'interpolation': True, 'masking': False}}, 'Smiles': {'type': 'text', 'column': 'smiles', 'target': True, 'preprocessor_arguments': {'tokenizer': 'smiles', 'tokenizer_regex': '(\\[[^\\]]+]|Br?|Cl?|N|O|S|P|F|I|b|c|n|o|s|p|\\(|\\)|\\.|=|#|-|\\+|\\\\\\\\|\\/|:|~|@|\\?|>|\\*|\\$|\\%[0-9]{2}|[0-9])'}}}
[2025-07-03 12:42:28,353][root][INFO] - Build dataset
[2025-07-03 12:42:28,354][root][INFO] - Loading existing preprocessor from: runs/pretraining/preprocessor.pkl
[2025-07-03 12:42:28,358][root][INFO] - Build preprocessors
[2025-07-03 12:42:29,522][root][INFO] - Build Datamodule
[2025-07-03 17:24:58,587][__main__][INFO] - Loading best Model from: runs/pretraining/version_1/checkpoints/epoch_46-step_45731.ckpt
[2025-07-03 17:30:25,439][analytical_fm.utils][INFO] - Top-1: 0.035
[2025-07-03 17:30:25,439][analytical_fm.utils][INFO] - Top-2: 0.056
[2025-07-03 17:30:25,439][analytical_fm.utils][INFO] - Top-3: 0.073
[2025-07-03 17:30:25,439][analytical_fm.utils][INFO] - Top-4: 0.087
[2025-07-03 17:30:25,439][analytical_fm.utils][INFO] - Top-5: 0.098
[2025-07-03 17:30:25,440][analytical_fm.utils][INFO] - Top-6: 0.105
[2025-07-03 17:30:25,440][analytical_fm.utils][INFO] - Top-7: 0.110
[2025-07-03 17:30:25,440][analytical_fm.utils][INFO] - Top-8: 0.114
[2025-07-03 17:30:25,440][analytical_fm.utils][INFO] - Top-9: 0.117
[2025-07-03 17:30:25,440][analytical_fm.utils][INFO] - Top-10: 0.119

```

Here are the screenshots of tensorboard:

![Image](https://github.com/user-attachments/assets/bf5aa9d5-0509-4379-bac4-ef80bfb99ba5)

It seams that `val_token_acc` has converged but `molecular_accuracy` is still very pool.

**Comment by pengjuRen99** [comment]
I also encountered the same problem. Although the validation set has converged, the effect of molecular prediction on the test set is very poor.

**Comment by Gankuaidian** [comment]
Same here, did u guys find out what's wrong?

**Comment by pengjuRen99** [comment]
I tried the latest code and this problem was solved.
